0003-Command_Definition_Syntax.md

```diff
@@ -0,0 +1,134 @@
+* Name: `Command_Definition_Syntax`
+* Date: 2021-09-20
+* Pull Request: [#24](https://github.com/opencomputeproject/Security/pull/24)
+
+# Objective
+
+Currently, the byte layouts of Cerberus protocol commands (hereafter "protocol
+structs") are defined ad-hoc by way of tables containing offsets. These tables
+present a number of problems:
+- They are painful to edit, because they are offset-based.
+- They are unnecessarily verbose in some cases.
+- They are not machine readable.
+
+This RFC proposes a formal syntax for the tables that is machine readable and
+concise in the common case.
+
+# Proposal
+
+Each table looks like the following in Markdown:
+```
+`message <Name>`
+| Type | Name | Description |
+| ---- | ---- | ----------- |
+...
+```
+Each table defines a message type. Each message type consists of a name and
+some number of fields. Each field consists of a type, a name, and an optional
+description.
+
+
+Message names must be be in `CamelCase`, with optional periods to separate
+components; field names must be in `snake_case`. All table cells, save the
+descriptions, are to be wrapped in backticks.
+
+The special field name `_` can be used for reserved fields.
+
+Descriptions should fit on one line and end in a period.
```

Comment:
So, I agree with this 100%, but I think we're forced into two unsavory options, because Markdown does not accommodate complex tables nicely...

- Use this rule.
- Allow switching to properly-indented HTML tables if absolutely required.

I'm fine with the latter if you are. If anything, it gives us an out while keeping the vast majority of tables readable in text form. Ultimately I think optimizing for the rendered form is not wrong.

Reply:
As discussed in mtg: we'll allow description lines to wrap beyond 80 cols if necessary, with all other lines of the table going out to 80 columns.